sdk/core/azure-core/src/test/java/com/azure/core/http/rest/PagedIterableTest.java

```diff
@@ -139,16 +139,8 @@ public void streamFirstPage() {
 
         assertEquals(pagedResponses.get(0), pagedIterable.streamByPage().limit(1).collect(Collectors.toList()).get(0));
 
-        /*
-         * The goal for this test would be that 0 next page retrieval calls are made but due to how Flux.concatWith
-         * works it needs to begin the next publisher to determine whether onNext or onComplete should trigger. This
-         * results in 2 next page retrieval calls for the following reason:
-         *
-         * - Makes the initial get first page call, then needs to validate that get next page emits. 1 call made.
-         * - Retrieving the first page in verification moves the stream iterator to the initial next page, Reactor then
-         *   needs to verify that the page after it emits. 2 calls made.
-         */
-        assertEquals(2, pagedFlux.getNextPageRetrievals());
+        // The goal for this test is that 0 next page retrieval calls are made.
+        assertEquals(0, pagedFlux.getNextPageRetrievals());
```

Comment:
@alzimmermsft - I've been thinking a bit to see how we can achieve the correct behavior here, i.e. avoiding 2 extra network calls consumer didn't ask us to perform, the new base type `ContinuablePagedFluxCore` will take care of this. I've updated the test to validate the expected next call count (i.e. 0). Please take a look and let me know if you've any concerns.

Reply:
I have no concerns, this is a great side affect of this change!

Reply:
Cool, thanks for the review Alan.

Reply:
and let's remove this FAQ entry once the PR is in: https://github.com/Azure/azure-sdk-for-java/wiki/Frequently-Asked-Questions#-why-does-synchronous-paged-iterator-fetches-2-pages-eagerly-when-iterating-by-page